builtin/logical/database/backend.go

```diff
@@ -78,6 +78,7 @@ func Backend(conf *logical.BackendConfig) *databaseBackend {
 		BackendType: logical.TypeLogical,
 	}
 
+	b.id = conf.BackendUUID
```

Comment:
I think instead of this you'd want to do the same thing we do for KV's encryption, just use the key for HMAC instead of encrypting. But we should get more opinions.